haskell/yesod/README.md

```diff
@@ -0,0 +1,118 @@
+# yesod docker image
+
+Use this image as the base image for your yesod based applications.
+
+## Usage
+
+The simplest usage is to create a `Dockerfile` for your project that looks like
+this:
+```
+FROM thoughtbot/yesod
+
+RUN mkdir -p /app
+WORKDIR /app
+
+COPY *.cabal ./
+RUN cabal install --dependencies-only -j4 --enable-tests
+```
+
+## Setup Your Yesod Project
+
+First, create a `Dockerfile` for your project. The one shown above is a good
+starting place.
+
+Test that your container builds before moving on:
+```
+docker build .
+```
+
+If the build succeeds, it will output a alpha-numeric string as the final
+container name. Use this name in the next step.
+
+Now, try to use the resulting container like so:
+
+```
+docker run -iv $PWD:/app CONTAINER_NAME yesod devel
+```
+
+Yesod should compile the app and error out because of a missing postgresql
+database. We can setup postgres or other database dependencies using
+[docker-compose].
+
+[docker-compose]: https://docs.docker.com/compose/install/
+
+Create a `docker-compose.yml` file inside your project. A simple setup might
+look like this:
+
+```
+db:
+  image: postgres:9.4
+  ports:
+   - "5432"
+web:
+  build: .
+  command: yesod devel
+  environment:
+    - HOST=0.0.0.0
+    - PGUSER=postgres
+    - PGPASS
+    - PGHOST=db
+    - PGPORT=5432
+    - PGDATABASE=postgres
```

Comment:
I think we want to leave this out so that it doesn't override the test database.

I created development and test databases in the container, the same we would for a normal application, using the default names Yesod looks for.